tests/test_processing_stack.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import time
+
+from pytest import mark
+
+APPLICATION_TAG_KEY = "ApplicationName"
+APPLICATION_TAG_VAL = "geospatial-data-lake"
+
+ENVIRONMENT_TAG_KEY = "EnvironmentType"
+ENVIRONMENT_TAG_VAL = os.environ.get("DEPLOY_ENV", "dev")
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def detect_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+    logger.info("Detected Compute Environments: %s", compute_environment_detection_response)
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+            break
+
+    return datalake_compute_environment
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
+    """
+    Trigger immediate launch of computing resources. This will significantly speed up Batch jobs
+    execution.
+    """
+    if (
+        int(datalake_compute_environment["computeResources"]["minvCpus"]) < 1
+        or int(datalake_compute_environment["computeResources"]["desiredvCpus"]) < 1
+    ):
+        batch_client.update_compute_environment(
+            computeEnvironment=datalake_compute_environment["computeEnvironmentName"],
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def detect_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+    logger.info("Detected State Machines: %s", state_machines_detection_response)
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {}
+        for tag in tags_detection_response["tags"]:
+            state_machine_tags[tag["key"]] = tag["value"]
```

Comment:
This can be simplified to a `dict` comprehension

```
state_machine_tags = {tag["key"]: tag["value"] for tag in tags_detection_response["tags"]}
```

Reply:
Done in 519d9039753ab1ad4eb1fbe52bad42533ca9a63b